simd/src/Kokkos_SIMD_AVX2.hpp

```diff
@@ -661,6 +661,35 @@ class simd<std::int32_t, simd_abi::avx2_fixed_size<4>> {
       _mm_add_epi32(static_cast<__m128i>(lhs), static_cast<__m128i>(rhs)));
 }
 
+KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION
+simd<std::int32_t, simd_abi::avx2_fixed_size<4>> operator>>(
+    simd<std::int32_t, simd_abi::avx2_fixed_size<4>> const& lhs,
+    int rhs) noexcept {
+  return simd<std::int32_t, simd_abi::avx2_fixed_size<4>>(
+      _mm_srai_epi32(static_cast<__m128i>(lhs), rhs));
+}
+KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION
+simd<std::int32_t, simd_abi::avx2_fixed_size<4>> operator>>(
+    simd<std::int32_t, simd_abi::avx2_fixed_size<4>> const& lhs,
+    simd<int, simd_abi::avx2_fixed_size<4>> const& rhs) noexcept {
+  return simd<std::int32_t, simd_abi::avx2_fixed_size<4>>(
+      _mm_srav_epi32(static_cast<__m128i>(lhs), static_cast<__m128i>(rhs)));
```

Comment:
Because this is for a signed type, I think we would want to use arithmetic right shift to keep shifted negative values still negative. Logical shift wouldn't do this since it fills in the leftmost bits with 0's.

Reply:
yeah: the standard says apply the operator. And << and >> preserve the sign when shifting negative values.